tutorials/nlp/Joint_Intent_and_Slot_Classification.ipynb

```diff
@@ -556,17 +556,251 @@
+    "print(config_file)\n",
+    "config = OmegaConf.load(config_file)\n",
+    "print(OmegaConf.to_yaml(config))"
+   ]
+  },
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {},
    "outputs": [],
-   "source": []
+   "source": [
+    "import time\n",
+    "\n",
+    "config.model.data_dir = f\"{DATA_DIR}/new_format\"\n",
+    "config.model.validation_ds.prefix = \"dev\"\n",
+    "config.model.test_ds.prefix = \"dev\"\n",
+    "config.model.class_balancing = \"weighted_loss\"\n",
+    "config.model.head.num_output_layers = 2\n",
+    "config.trainer.max_epochs = 40\n",
+    "run_name = \"test\"\n",
+    "\n",
+    "# checks if we have GPU available and uses it\n",
+    "cuda = 1 if torch.cuda.is_available() else 0\n",
+    "config.trainer.gpus = cuda\n",
+    "\n",
+    "# config.trainer.precision = 16 if torch.cuda.is_available() else 32\n",
+    "config.trainer.precision = 32\n",
+    "\n",
+    "# remove distributed training flags\n",
+    "config.trainer.accelerator = None\n",
+    "\n",
+    "trainer = pl.Trainer(**config.trainer)\n",
+    "config.exp_manager.exp_dir = os.path.join(DATA_DIR, \"output/\" + run_name)\n",
+    "config.exp_manager.create_checkpoint_callback = True\n",
+    "config.exp_manager.version = time.strftime('%Y-%m-%d_%H-%M-%S')\n",
+    "\n",
+    "exp_dir = exp_manager(trainer, config.get(\"exp_manager\", None))\n",
+    "model = nemo_nlp.models.MultiLabelIntentSlotClassificationModel(config.model, trainer=trainer)\n",
+    "trainer.fit(model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Evaluation"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To see how the model performs, we can evaluate the performance of the trained model on a test data file. Here we would load the best checkpoint (the one with the lowest validation loss) and create a model (eval_model) from the checkpoint. We will use the same trainer for testing. We will need to use the file specified in the output during the training process."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# specify checkpoint path with .nemo file (Look for /MultiLabelIntentSlot.nemo)\n",
+    "checkpoint_path = \"\"\n",
+    "\n",
+    "# load the model from this checkpoint\n",
+    "eval_model =  nemo_nlp.models.MultiLabelIntentSlotClassificationModel.restore_from(checkpoint_path)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Optimizing Threshold\n",
+    "Before using our model, we find the probability rounding threshold that gives us the best results on the validation set. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "eval_model.optimize_threshold(config.model.test_ds)"
```

Comment:
After running `optimize_threshold`, I think it would be good to add a cell where you call `eval_model.threshold` so the user can see that the "best" threshold, based on F1 scores, was set as a model attribute. Also please explain that in the notebook.